velox/core/PlanFragment.h

```diff
@@ -42,10 +43,22 @@ struct PlanFragment {
   ExecutionStrategy executionStrategy{ExecutionStrategy::kUngrouped};
   int numSplitGroups{0};
 
+  /// Contains plan nodes that need to be executed in the grouped mode.
+  std::unordered_set<PlanNodeId> groupedExecutionScanNodeIds;
+
+  /// Returns true if the fragment uses grouped execution strategy meaning that
+  /// at least one pipeline has a node that should run grouped execution.
+  /// Note that it does not mean that all pipelines run grouped execution.
   inline bool isGroupedExecution() const {
     return executionStrategy == ExecutionStrategy::kGrouped;
   }
 
+  /// Returns true for nodes that use grouped execution, false otherwise.
```

Comment:
This is misleading since a non-source node can be part of the pipeline that runs in "grouped" mode. It would be nice to clarify.

Reply:
@mbasmanova 
What would you like to see in this comment?
Asking this because I want to reduce the number of iterations we spend on this change. :)

We could start from

`/// Returns true for leaf nodes that use grouped execution, false otherwise.`

Would that be clearer?

Reply:
Also, do we want to consider a different function name as well?

Reply:
Yes, renaming a function would be helpful too. Can you share some context for this change. I assume it is a small part of a larger change, but I don't know that larger change is.

Reply:
Synced offline.